validator/src/cli/thread_args.rs

```diff
@@ -316,28 +310,6 @@ impl ThreadArg for ReplayTransactionsThreadsArg {
     }
 }
 
-pub struct RocksdbCompactionThreadsArg;
-impl ThreadArg for RocksdbCompactionThreadsArg {
-    const NAME: &'static str = "rocksdb_compaction_threads";
-    const LONG_NAME: &'static str = "rocksdb-compaction-threads";
-    const HELP: &'static str = "Number of threads to use for rocksdb (Blockstore) compactions";
-
-    fn default() -> usize {
-        solana_ledger::blockstore::default_num_compaction_threads().get()
-    }
-}
-
-pub struct RocksdbFlushThreadsArg;
-impl ThreadArg for RocksdbFlushThreadsArg {
-    const NAME: &'static str = "rocksdb_flush_threads";
-    const LONG_NAME: &'static str = "rocksdb-flush-threads";
-    const HELP: &'static str = "Number of threads to use for rocksdb (Blockstore) memtable flushes";
-
-    fn default() -> usize {
-        solana_ledger::blockstore::default_num_flush_threads().get()
-    }
-}
```

Comment:
The purpose of this function is to have uniform behavior between all the different arguments that control thread counts. I think we should move this over to the blockstore function instead of recreating the logic.

I could see creating a new function similar to:
```rust
fn new_thread_arg<'a, T: ThreadArg>(default: &str) -> Arg<'_, 'a> {
```
but does NOT accept any arguments (ie the `default: &str`). This would allow us to gradually shift away from the `DefaultThreadArgs` struct, and then eventually kill the function that takes a `&str`